packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -3715,14 +3718,11 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin, Res
       );
       // Pageless routes will be recorded when we update the middle of the old
       // list.
-      if (!oldEntry.hasPage) {
+      if (!oldEntry.pageBased) {
         continue;
       }
 
-      assert(oldEntry.hasPage);
```

Comment:
Why remove this?

Reply:
I think this check is redundant  given the if case above should already short-circuited pageless route